include/golioth/ota.h

```diff
@@ -68,6 +68,14 @@ enum golioth_ota_reason
     GOLIOTH_OTA_REASON_FIRMWARE_UPDATE_FAILED,
     /// Protocol not supported
     GOLIOTH_OTA_REASON_UNSUPPORTED_PROTOCOL,
+    /// IO error while trying to store component
+    GOLIOTH_OTA_REASON_IO,
+    /// Failure to download block
+    GOLIOTH_OTA_REASON_BLOCK_FAILED,
+    /// Resuming the interrupted download
+    GOLIOTH_OTA_REASON_RESUMING,
```

Comment:
I'm curious if changing reason in this case is worth it. For example, if we are downloading 100 blocks and every 5th block download fails, it might create more noise than is helpful. I think the key thing we want to communicate is:
1. If we are downloading, are we progressing?
2. If we are not progressing, why?

If a block fails and it is retry-able, then I think I would still consider that progressing, until we eventually got into backoff due to repeated failures (then we would move to `AWAIT_RETRY`). If we want to communicate how often that a block is getting re-requested (i.e. network stats), that is something we could observe and report on server-side without needing to have the device report it. Do you think that would accomplish your goals here?

Reply:
I think it would accomplish the same goal. There is something to be said for having the actual report from the device to know its state. I think you're asking due to how the history will display on the console?

We're trying to communicate how often the block download is resuming as not doing so would mask spotty connection and lead to questions about data usage. We need to send alternating messages to avoid deduplication on the server from masking repeated messages.

Reply:
> I think you're asking due to how the history will display on the console?

I'm actually less concerned about presentation -- presentation can be adapted to most effectively present useful information based on the data that we have. I'm more concerned about reporting more data on an already spotty connection. If the goal is communicating spotty connection statistics then I think we could do as good or better using server-side metrics and / or supporting reporting statistics from a device after an OTA download completes.